plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMySqlTypeMapping.java

```diff
@@ -482,115 +482,7 @@ public void testDate()
-                .addRoundTrip("TIMESTAMP '1970-01-01 00:00:00.9'", "TIMESTAMP '1970-01-01 01:00:00.900'")
-                .addRoundTrip("TIMESTAMP '1970-01-01 00:00:00.123'", "TIMESTAMP '1970-01-01 01:00:00.123'")
-                .addRoundTrip("TIMESTAMP '1970-01-01 00:00:00.123000'", "TIMESTAMP '1970-01-01 01:00:00.123'")
-                .addRoundTrip("TIMESTAMP '1970-01-01 00:00:00.999'", "TIMESTAMP '1970-01-01 01:00:00.999'")
-                // max supported precision
-                .addRoundTrip("TIMESTAMP '1970-01-01 00:00:00.123456'", "TIMESTAMP '1970-01-01 01:00:00.123'")
-
-                .addRoundTrip("TIMESTAMP '2020-09-27 12:34:56.1'", "TIMESTAMP '2020-09-27 12:34:56.100'")
-                .addRoundTrip("TIMESTAMP '2020-09-27 12:34:56.9'", "TIMESTAMP '2020-09-27 12:34:56.900'")
-                .addRoundTrip("TIMESTAMP '2020-09-27 12:34:56.123'", "TIMESTAMP '2020-09-27 12:34:56.123'")
-                .addRoundTrip("TIMESTAMP '2020-09-27 12:34:56.123000'", "TIMESTAMP '2020-09-27 12:34:56.123'")
-                .addRoundTrip("TIMESTAMP '2020-09-27 12:34:56.999'", "TIMESTAMP '2020-09-27 12:34:56.999'")
-                // max supported precision
-                .addRoundTrip("TIMESTAMP '2020-09-27 12:34:56.123456'", "TIMESTAMP '2020-09-27 12:34:56.123'")
-
-                // round down
-                .addRoundTrip("TIMESTAMP '1970-01-01 00:00:00.123451'", "TIMESTAMP '1970-01-01 01:00:00.123'")
-
-                // nanos round up, end result rounds down
-                .addRoundTrip("TIMESTAMP '1970-01-01 00:00:00.123499'", "TIMESTAMP '1970-01-01 01:00:00.123'")
-                .addRoundTrip("TIMESTAMP '1970-01-01 00:00:00.123399'", "TIMESTAMP '1970-01-01 01:00:00.123'")
-
-                // round up
-                .addRoundTrip("TIMESTAMP '1970-01-01 00:00:00.123999'", "TIMESTAMP '1970-01-01 01:00:00.124'")
-
-                // max precision
-                .addRoundTrip("TIMESTAMP '1970-01-01 00:00:00.123456'", "TIMESTAMP '1970-01-01 01:00:00.123'")
-
-                // round up to next second
-                .addRoundTrip("TIMESTAMP '1970-01-01 00:00:00.999999'", "TIMESTAMP '1970-01-01 01:00:01.000'")
-
-                // round up to next day
-                .addRoundTrip("TIMESTAMP '1970-01-01 23:59:59.999999'", "TIMESTAMP '1970-01-02 00:00:00.000'")
-
-                // negative epoch - round up
-                .addRoundTrip("TIMESTAMP '1969-12-31 23:59:59.999995'", "TIMESTAMP '1970-01-01 00:00:00.000'")
-                .addRoundTrip("TIMESTAMP '1969-12-31 23:59:59.999949'", "TIMESTAMP '1970-01-01 00:00:00.000'")
-                .addRoundTrip("TIMESTAMP '1969-12-31 23:59:59.999994'", "TIMESTAMP '1970-01-01 00:00:00.000'")
-
-                .execute(getQueryRunner(), session, mysqlCreateAndInsert("tpch.test_timestamp"));
-    }
-
-    @Test(dataProvider = "sessionZonesDataProvider")
-    public void testTimestamp(ZoneId sessionZone)
+    public void testTimestampFromTrino(ZoneId sessionZone)
```

Comment:
I prefer to have "their timestamp" test before "our timestamp" test, because the former exercises read path, and the latter both read and write paths.

If we kept original method ordering, would the diff be smaller?
i.e. what's acually changed here, and what's just moved?

Reply:
> I prefer to have "their timestamp" test before "our timestamp" test

Oh, it turns out doing that also makes the diff a lot, so I'll do that.

Reply:
Yeah, looking at the new diff, this way is _much_ better.